source/HL_i2c.c

```diff
@@ -140,18 +141,18 @@ void i2cInit(void)
                  | (uint32)(1U);     /* scl pin */
 
     /** - set interrupt enable */
-    i2cREG1->IMR    = (uint32)((uint32)0U << 6U)     /* Address as slave interrupt      */
-                    | (uint32)((uint32)0U << 5U)     /* Stop Condition detect interrupt */
-                    | (uint32)((uint32)0U << 4U)     /* Transmit data ready interrupt   */
-                    | (uint32)((uint32)0U << 3U)     /* Receive data ready interrupt    */
-                    | (uint32)((uint32)0U << 2U)     /* Register Access ready interrupt */
-                    | (uint32)((uint32)0U << 1U)     /* No Acknowledgment interrupt    */
-                    | (uint32)((uint32)0U);     /* Arbitration Lost interrupt      */
+    i2cREG1->IMR    = (uint32)((uint32)1U << 6U)     /* Address as slave interrupt      */
+                    | (uint32)((uint32)1U << 5U)     /* Stop Condition detect interrupt */
+                    | (uint32)((uint32)1U << 4U)     /* Transmit data ready interrupt   */
+                    | (uint32)((uint32)1U << 3U)     /* Receive data ready interrupt    */
+                    | (uint32)((uint32)1U << 2U)     /* Register Access ready interrupt */
+                    | (uint32)((uint32)1U << 1U)     /* No Acknowledgment interrupt    */
+                    | (uint32)((uint32)1U);     /* Arbitration Lost interrupt      */
 
     i2cREG1->MDR |= (uint32)I2C_RESET_OUT; /* i2c out of reset */
 
     /** - initialize global transfer variables */
-    g_i2cTransfer_t[0U].mode   = (uint32)0U << 4U;
+    g_i2cTransfer_t[0U].mode   = (uint32)1U << 4U;

```

Comment:
No need to OR in the value?

Reply:
If not, perhaps a more descriptive comment...

Reply:
This is auto-generated. Nothing I can do unfortunately